tests/unit/test_weighting.py

```diff
@@ -0,0 +1,37 @@
+from src.core.weighting import weighting_operation
+import pytest
+
+from src.core.exceptions import InvalidWeightSize
+
+
+# @pytest.mark.parametrize()
+# def test_weighting_operation():
+#     list_whatever = function_asda()
+#     tensor = create_sc_tensor(list_whatever, 1)
+#     res = operation_ponderacao(tensor)
+#     assert res == [[[0.23234132]], [[0.00202036]], [[0.30709461]]]
+
+
+def test_weighting_operation():
+    df = [0.30, 0.32, 0.12]
+    weights = [0.30, 0.45, 0.25]
+    res = weighting_operation(df, weights)
+
+    assert pytest.approx(res.tolist()) == [0.09, 0.144, 0.03]
+
+
+def test_weighting_operation_2():
+    df = [0.40, 0.50, 0.10]
+    weights = [0.30, 0.45, 0.25]
+    res = weighting_operation(df, weights)
+
+    assert pytest.approx(res.tolist()) == [0.12, 0.225, 0.025]
+
+
+def test_diference_size():
+    df = [0.40, 0.40, 0.10, 0.10]
+    weights = [0.30, 0.45, 0.25]
+    try:
+        weighting_operation(df, weights)
+    except InvalidWeightSize:
+        assert True
```

Comment:
Isso aqui funciona, mas não é a melhor maneira de fazer isso. Vejam os seguintes links:

- https://stackoverflow.com/questions/23337471/how-to-properly-assert-that-an-exception-gets-raised-in-pytest
- https://docs.pytest.org/en/6.2.x/assert.html#assertions-about-expected-exceptions
- https://github.com/fga-eps-mds/2021-2-MeasureSoftGram-Core/blob/develop/tests/unit/test_interpretation_functions.py